editor/script_editor_debugger.cpp

```diff
@@ -1476,13 +1477,27 @@ void ScriptEditorDebugger::start() {
 		perf_max.write[i] = 0;
 	}
 
-	int remote_port = (int)EditorSettings::get_singleton()->get("network/debug/remote_port");
-	if (server->listen(remote_port) != OK) {
+	const int max_tries = 6;
+	remote_port = (int)EditorSettings::get_singleton()->get("network/debug/remote_port");
+	int current_try = 0;
+	// Find first available port.
+	Error err = server->listen(remote_port);
+	while (err != OK && current_try < max_tries) {
+		EditorNode::get_log()->add_message(String("Remote debugger failed listening on port: ") + itos(remote_port) + String(" Retrying on new port: " + itos(remote_port + 1)), EditorLog::MSG_TYPE_WARNING);
+		current_try++;
+		remote_port++;
+		OS::get_singleton()->delay_usec(1000);
+		err = server->listen(remote_port);
+	}
+	// No suitable port found.
+	if (err != OK) {
 		EditorNode::get_log()->add_message(String("Error listening on port ") + itos(remote_port), EditorLog::MSG_TYPE_ERROR);
-		return;
+		EditorNode::get_log()->add_message(String("Remote debugger error listening for connections. No free port"), EditorLog::MSG_TYPE_ERROR);
 	}
-
+	// Success.
+	EditorNode::get_log()->add_message("--- Debugging process started ---", EditorLog::MSG_TYPE_EDITOR);
```

Comment:
This line should not be here, it's printed when the client connects (causing it to be printed twice).